### ISSUE TITLE
Maven build: Annotations with RetentionPolicy = SOURCE are omitted by JvmAnnotationTarget.getAnnotations()

### ISSUE THREAD

**Issue by OLibutzki** [eclipse/xtext-extras#446]
From a Xbase DSL I reference a Java type which is annotated with an annotation which's RetentionPolicy is SOURCE. The Java type is located in the same maven module as the dsl file which references it.

Within Eclipse the getAnnotations() method of the referenced java types also returns the annotations with RetentionPolicy = SOURCE.

I would expect this behaviour if the referenced Java type is located in another Maven module as the Source-Annotations are discarded by the compiler. But in my case the DSL file and the reference Java class are located in the same module and therefore the source file is available.

**Comment by OLibutzki** [comment]
Xtext version: 2.17.1

**Comment by OLibutzki** [comment]
To illustrate the issue I created https://github.com/OLibutzki/XtextExtras446

It's a very simple DSL. You can just reference a JvmType. The ModelInferrer generates a class and a `isMySourceAnnotationPresent()` for each dsl file.
The inferrer checks for the existence of `de.libutzki.xtextextras446.MySourceAnnotation` at the referenced type and generates `return true;` as the method body is the annotation is found. If the annotation is not found, false is returned.

I also added a small runtime project which uses the DSL. Within Eclipse executing `de.libutzki.xtextextras446.XtextExtras446Test` succeeds, but running this with Maven fails as the annotation seems to not be visible to the inferrer.

I hope this helps you to reproduce the issue.

Kind regards
Oliver

**Comment by cdietrich** [comment]
reproducing is one thing. but i dont really have an idea how to fix.

**Comment by OLibutzki** [comment]
I see, but you share my expectation that the annotation should be visible/accessible?

**Comment by cdietrich** [comment]
well i am not sure since there is not a proper contract and our jdt integration is very bogus in that regards (e.g. container handling)
@szarnekow what do you think?

**Comment by cdietrich** [comment]
@OLibutzki do you make use of javasourcelanguage in maven?

**Comment by cdietrich** [comment]
hmm that does actually not work neither

```
[ERROR] Failed to execute goal org.eclipse.xtext:xtext-maven-plugin:2.17.1:generate (default) on project xtextextras446-runtime: Execution default of goal org.eclipse.xtext:xtext-maven-plugin:2.17.1:generate failed: no index installed -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.eclipse.xtext:xtext-maven-plugin:2.17.1:generate (default) on project xtextextras446-runtime: Execution default of goal org.eclipse.xtext:xtext-maven-plugin:2.17.1:generate failed: no index installed
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:215)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
```

**Comment by OLibutzki** [comment]
> @OLibutzki you do notmake use of javasourcelanguage in maven, what happens if you do?

I am not aware of this. Do you talk about the `javaSupport` flag at `org.eclipse.xtext.maven.Language`? It  seems to be true by default.

**Comment by cdietrich** [comment]
no there is xtext.java maven/gralde bundle and 
`<setup>org.eclipse.xtext.java.JavaSourceLanguageSetup</setup>`
but that one does not seem to work with maven plugin.

**Comment by szarnekow** [comment]
> => I'd rather see it as a bug that they are seen in eclipse than that they are not seen in maven
> 
> @szarnekow what do you think?

Given that both sources reside in the same maven module, it would be nice if the annotations with source-retention are still present. To me this is a missing feature of the integration with Java when building from cmd line.

**Comment by cdietrich** [comment]
hmmm as we read class files this is not easily fixable correct?

**Comment by szarnekow** [comment]
> hmmm as we read class files this is not easily fixable correct?

True. It would mean an overhaul of the Java integration to be sported from source code instead of byte code.

**Comment by OLibutzki** [comment]
... and in Eclipse you rely on the source files?

**Comment by cdietrich** [comment]
no we rely on jdt

**Comment by OLibutzki** [comment]
Ok, but the information that this enhancement cannot be made easily helps me. I will try to circumvent this issue. Thank you both!
